nixos/modules/services/hardware/acpid.nix

```diff
@@ -3,153 +3,189 @@
+    mkdir -p $out/scripts
+    mkdir -p $out/handlers
+    ${concatStringsSep "\n" (mapAttrsToList mkScript handlers)}
+    ${concatStringsSep "\n" (mapAttrsToList mkHandler handlers)}
+  '';
+
+  acpiConfFor = cfg: let
+    canonicalHandlers = {
+      powerEvent = {
+        event = "button/power.*";
+        action = cfg.powerEventCommands;
+      };
 
-  options = {
+      lidEvent = {
+        event = "button/lid.*";
+        action = cfg.lidEventCommands;
+      };
 
-    services.acpid = {
+      acEvent = {
+        event = "ac_adapter.*";
+        action = cfg.acEventCommands;
+      };
+    };
+  in mkAcpiHandlers ((lib.filterAttrs (_: handler: handler.action != "") canonicalHandlers) // cfg.handlers);
 
-      enable = mkEnableOption (lib.mdDoc "the ACPI daemon");
+  options = stage2: {
+    enable = mkEnableOption (lib.mdDoc "the ACPI daemon");
+    package = mkPackageOption pkgs "acpid" {};
 
-      logEvents = mkOption {
-        type = types.bool;
-        default = false;
-        description = lib.mdDoc "Log all event activity.";
-      };
+    logEvents = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc "Log all event activity";
+    };
 
-      handlers = mkOption {
-        type = types.attrsOf (types.submodule {
-          options = {
-            event = mkOption {
-              type = types.str;
-              example = literalExpression ''"button/power.*" "button/lid.*" "ac_adapter.*" "button/mute.*" "button/volumedown.*" "cd/play.*" "cd/next.*"'';
-              description = lib.mdDoc "Event type.";
-            };
-
-            action = mkOption {
-              type = types.lines;
-              description = lib.mdDoc "Shell commands to execute when the event is triggered.";
-            };
-          };
-        });
-
-        description = lib.mdDoc ''
-          Event handlers.
-
-          ::: {.note}
-          Handler can be a single command.
-          :::
-        '';
-        default = {};
-        example = {
-          ac-power = {
-            event = "ac_adapter/*";
-            action = ''
-              vals=($1)  # space separated string to array of multiple values
-              case ''${vals[3]} in
-                  00000000)
-                      echo unplugged >> /tmp/acpi.log
-                      ;;
-                  00000001)
-                      echo plugged in >> /tmp/acpi.log
-                      ;;
-                  *)
-                      echo unknown >> /tmp/acpi.log
-                      ;;
-              esac
+    handlers = mkOption {
+      type = types.attrsOf (types.submodule {
+        options = {
+          event = mkOption {
+            type = types.str;
+            description = lib.mdDoc (''
+              Event type
+            '' + lib.optionalString (!stage2) ''
+
+              ::: {.note}
+              Many events will require kernel modules to be available via `boot.initrd.availableKernelModules`.
```

Comment:
Are there any other useful information we can give or link how to know which ones are required?

Reply:
This is what I found https://wiki.archlinux.org/title/ACPI_modules